### ISSUE TITLE
bq_project_query() compatibility with glue::glue_sql()

### ISSUE THREAD

**Issue by grantmcdermott** [r-dbi/bigrquery#334]
This may be a question for the [glue](https://github.com/tidyverse/glue) package, but let me try here first.

I often use `glue::glue_sql()` to compose lengthy or multipart SQL queries. This approach works very well with bigrquery when you want to download data to your R session. However, I've been unable to combine it with `bq_project_query()` to actually save the resulting table _directly_ to BigQuery. 

Here's a reprex adapted from an example in the `bq_project_query()` help documentation.

``` r
library(bigrquery)
library(DBI)
library(glue)

## Get GCP billing ID as environment variable
proj_id = Sys.getenv("GCE_DEFAULT_PROJECT_ID")

con <- 
  dbConnect(
    bigrquery::bigquery(),
    project = "publicdata",
    dataset = "samples",
    billing = proj_id
  )

natality <- "natality"

sql_query <-
  glue_sql(
  "
  SELECT
  count(*)
  FROM
  {`natality`}
  ",
    .con = con
  )

## Not run, but it's easy to show that the query itself is valid
# dbGetQuery(con, sql_query)

## Fails to write the resulting table to BigQuery
bq_project_query(
  proj_id,
  sql_query
  )
#> Error: Table name "natality" missing dataset while no default dataset is set in the request. [invalid]
```

<sup>Created on 2019-07-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

So basically, `bq_project_query()` is parsing the "natality" table/object without the full dataset prefix that `glue_sql()` records separately as the connection object (i.e. "publicdata.samples."). There might be an easy way to parse these things together, but I'm afraid that's beyond my current know-how.

**Comment by hadley** [comment]
Can't you just spell out the full table name in the SQL you're generating?
